libs/src/api/base.ts

```diff
@@ -0,0 +1,180 @@
+import type { Comstock } from '../services/comstock'
+import type { CreatorNode } from '../services/creatorNode'
+import type { AudiusContracts } from '../services/dataContracts'
+import type { DiscoveryProvider } from '../services/discoveryProvider'
+import type { EthContracts } from '../services/ethContracts'
+import type { WormholeClient } from '../services/ethContracts/WormholeClient'
+import type { EthWeb3Manager } from '../services/ethWeb3Manager'
+import type { Hedgehog } from '../services/hedgehog'
+import type { IdentityService } from '../services/identity'
+import type { SolanaAudiusData } from '../services/solanaAudiusData'
+import type SolanaWeb3Manager from '../services/solanaWeb3Manager'
+import type { Web3Manager } from '../services/web3Manager'
+import type { UserStateManager } from '../userStateManager'
+import type { Captcha } from '../utils'
+
+export const Services = Object.freeze({
+  IDENTITY_SERVICE: 'Identity Service',
+  HEDGEHOG: 'Hedgehog',
+  DISCOVERY_PROVIDER: 'Discovery Provider',
+  CREATOR_NODE: 'Creator Node',
+  COMSTOCK: 'Comstock',
+  SOLANA_WEB3_MANAGER: 'Solana Web3 Manager'
+})
+
+export type BaseConstructorArgs = [
```

Comment:
Kinda weird – had to export this for use in child classes that have `constructor(...args)` style constructors. Really should be all object based but bigger change than I wanted to make